types/api.d.ts

```diff
@@ -0,0 +1,104 @@
+export default API;
+export type nHentaiOptions = import("./options").nHentaiOptions;
+/**
+ * API arguments
+ */
+export type APIArgs = {
+    /**
+     * API host.
+     */
+    host: string;
+    /**
+     * API endpoint URL path generator.
+     */
+    apiPath: Function;
+};
+/**
+ * Class used for interaction with nHentai API.
+ * @class
+ */
+declare class API {
+    /**
+     * API path class
+     * @type {APIPath}
+     * @static
+     * @private
+     */
+    private static APIPath;
+    /**
+     * Applies provided options on top of defaults.
+     * @param {nHentaiOptions} options Options to apply.
+     */
+    constructor(options: nHentaiOptions);
```

Comment:
Looks like it would work with no arguments.
Actually that's auto generated declarations from jsdoc;

Reply:
> Looks like it would work with no arguments. Actually that's auto generated declarations from jsdoc;

Indeed so it should be an optional param because currently, TypeScript throws an error when the argument isn't provided and it's kinda annoying for me.

Reply:
fixed